grisera_api/graph_api_service.py

```diff
@@ -53,10 +53,37 @@ def create_properties(self, node_id: int, node_model: BaseModel):
         """
         node_dict = node_model.dict()
         request_body = [{"key": key, "value": value} for key, value in node_dict.items()
-                        if value is not None and key != 'additional_properties']
+                        if value is not None and key != 'additional_properties' and key != 'authors' and key != 'publication']
 
-        if node_dict['additional_properties'] is not None:
-            [request_body.append({"key": property['key'], "value": property['value']})
-             for property in node_dict['additional_properties']]
+        if 'additional_properties' in node_dict and node_dict['additional_properties'] is not None:
+            [request_body.append({"key": additional_properties['key'], "value": additional_properties['value']})
+             for additional_properties in node_dict['additional_properties']]
+
+        if 'authors' in node_dict and node_dict['authors'] is not None:
+            for authors in node_dict['authors']:
+                request_body.append({"key": "name", "value": authors['name']})
+                request_body.append({"key": "institution", "value": authors['institution']})
+
+        if 'publication' in node_dict and node_dict['publication'] is not None:
+            publication = node_dict['publication']
+            request_body.append({"key": "title", "value": publication['title']})
+            for authors in node_dict['authors']:
```

Comment:
I'm not sure if I follow... Doesn't this mean that info about authors will be added twice?

Reply:
I thought that there can be different authors for the publication and for the experiment. That is why the info is added for experiment's authors and for publication's authors.

Reply:
Got it :smile: